backend/module/chaos/serverexperimentation/xds/ecds.go

```diff
@@ -0,0 +1,255 @@
+package xds
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	gcpCoreV3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	gcpRoute "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	gcpFilterCommon "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/common/fault/v3"
+	gcpFilterFault "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/fault/v3"
+	gcpType "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	gcpTypes "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	proto "github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/duration"
+	"go.uber.org/zap"
+
+	experimentation "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	serverexperimentation "github.com/lyft/clutch/backend/api/chaos/serverexperimentation/v1"
+)
+
+const (
+	EnvoyDownstreamServiceClusterHeader   = `x-envoy-downstream-service-cluster`
+	FaultFilterConfigNameForExternalFault = `envoy.egress.extension_config.%s`
+	FaultFilterConfigNameForInternalFault = `envoy.extension_config`
+
+	faultFilterTypeURL = `type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault`
+
+	delayPercentRuntime    = `ecds_runtime_override_do_not_use.http.delay.percentage`
+	delayDurationRuntime   = `ecds_runtime_override_do_not_use.http.delay.fixed_duration_ms`
+	abortHttpStatusRuntime = `ecds_runtime_override_do_not_use.http.abort.http_status`
+	abortPercentRuntime    = `ecds_runtime_override_do_not_use.http.abort.abort_percent`
+)
+
+// Default abort and delay fault configs
+var DefaultAbortFaultConfig = &gcpFilterFault.FaultAbort{
+	ErrorType: &gcpFilterFault.FaultAbort_HttpStatus{
+		HttpStatus: 503,
+	},
+	Percentage: &gcpType.FractionalPercent{
+		Numerator:   0,
+		Denominator: gcpType.FractionalPercent_HUNDRED,
+	},
+}
+
+var DefaultDelayFaultConfig = &gcpFilterCommon.FaultDelay{
+	FaultDelaySecifier: &gcpFilterCommon.FaultDelay_FixedDelay{
+		FixedDelay: &duration.Duration{
+			Nanos: 1000000, // 0.001 second
+		},
+	},
+	Percentage: &gcpType.FractionalPercent{
+		Numerator:   0,
+		Denominator: gcpType.FractionalPercent_HUNDRED,
+	},
+}
+
+type SafeEcdsResourceMap struct {
+	mu                    sync.Mutex
+	requestedResourcesMap map[string][]string
+}
+
+type ECDSConfig struct {
+	enabledClusters map[string]struct{}
+
+	ecdsResourceMap *SafeEcdsResourceMap
+}
+
+func generateECDSResource(experiments []*experimentation.Experiment, ttl *time.Duration, logger *zap.SugaredLogger) []gcpTypes.ResourceWithTtl {
+	var downstreamCluster string
+	var upstreamCluster string
+	var isExternalFault = false
+
+	abort := DefaultAbortFaultConfig
+	delay := DefaultDelayFaultConfig
```

Comment:
Seems like createAbortDelay already handles this default logic, so I don't think we need these?

Reply:
This is just a safety mechanism incase `experiments` are empty. In that case, we need some default